src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,115 @@ void medViewContainer::removeColorIndicator(QColor color)
 {
     d->poolIndicator->removeColorIndicator(color);
 }
+
+/**
+ * @brief saves the scene in a XML file
+ * Saves views (all layers), and toolboxes parameters
+ * 	expected tree is as follow:
+ *  workingDir (user-defined)
+ * 		|-viewID0
+ * 			|-mapping.xml
+ * 			|-layerID01.xml
+ * 			|-layerID02.xml
+ * 			|-data_file (as saved by the dedicated writer)
+ */
+QString medViewContainer::saveScene()
+{
+    QString dirPath = QFileDialog::getExistingDirectory(this, tr("Open Directory"),"/home",QFileDialog::ShowDirsOnly);
+    QDir workingDir(dirPath);
+    QDomDocument doc("xml");
+    QDomElement root=doc.createElement("scene");
+    doc.appendChild(root);
+
+    QString subDirName;
+
+    medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
+    if (layeredView!=0)
+    {
+        subDirName="view"+QUuid::createUuid ().toString();
+        if(!workingDir.mkdir(subDirName))
+            qWarning() << " failed to create new directory: "<<subDirName;
```

Comment:
If the warning is thrown because the directory has not been created, that's usually because the application doesn't have the access rights to create a directory in that location. The user can fix this by elevating the privilege of the application (run as admin in windows, sudo in linux).

I would personally say then that you should have a popup when it fails in this case. Up to you though!